crates/l2/proposer/l1_committer.rs

```
@@ -276,16 +276,20 @@ impl Committer {
                 None => 0,
             };
 
+            let new_nonce = if account_update.info.is_none() {
+                prev_nonce
+            } else {
+                account_update
+                    .info
+                    .as_ref()
+                    .ok_or(CommitterError::FailedToRetrieveDataFromStorage)?
+                    .nonce
+            };
             modified_accounts.insert(
                 account_update.address,
                 AccountStateDiff {
                     new_balance: account_update.info.clone().map(|info| info.balance),
-                    nonce_diff: (account_update
-                        .info
-                        .clone()
-                        .ok_or(CommitterError::FailedToRetrieveDataFromStorage)?
-                        .nonce
-                        - prev_nonce)
+                    nonce_diff: (new_nonce - prev_nonce)
```

It should never happen, but we should check for overflow here.

For that, I suggest using [overflowing_sub](https://doc.rust-lang.org/std/primitive.u64.html#method.overflowing_sub) and maybe log a warning.

---

What do you think of this approach Fran? https://github.com/lambdaclass/ethrex/pull/2133/commits/638f29bacc54af204c4018119ab67ed26bf7daa4

I went for `checked_sub` instead because we use more often in the codebase, plus something about `overflowing_sub` seems fishy to me. 
Isn't it better to raise an error rather than commit an invalid/erroneous nonce?

---

My bad, I got the mixed up. `checked_sub` is the correct one to use.